### ISSUE TITLE
Upload Symbols Failed: Missing dSYM for GiphyUISDK.framework in Xcode 16+

### ISSUE THREAD

**Issue by JCsplash** [Giphy/giphy-ios-sdk#264]
## 🐛 Bug Report

When attempting to submit an archived build to the App Store using Xcode 16+ (e.g., Xcode 16.2), the submission process gives the following warning:

```
Upload Symbols Failed
The archive did not include a dSYM for the GiphyUISDK.framework with the UUIDs [SOME-UUID]. Ensure that the archive's dSYM folder includes a DWARF file for GiphyUISDK.framework with the expected UUIDs.
```

## To Reproduce

Steps to Reproduce:

1. Archive the project using Xcode 16+.
2. Attempt to upload the build to the App Store.
3. Observe the “Upload Symbols Failed” error.

## Expected behavior

The archive should include the dSYM for GiphyUISDK.framework with the expected UUIDs, allowing the build to upload successfully.

## Actual Behavior

The dSYM for GiphyUISDK.framework is missing from the archive when using Xcode 16+, causing the upload to fail.

## Additional Notes
Works as expected with Xcode 15 and earlier.
GiphyUISDK.framework is integrated using CocoaPods (installed Giphy v2.2.13).
Tested with clean builds and verified that dSYM generation settings are correct.

## Your Environment

- Giphy iOS SDK version: 2.2.13
- Xcode Version: Xcode 16.2

**Comment by vsp0697** [comment]
@JCsplash, I am facing the same issue. Could you please share how you resolved it? Any suggestions or solutions would be greatly appreciated. Thank you!

**Comment by JCsplash** [comment]
@vsp0697 We never solved this sadly. So we had to keep using Xcode 15. I hope an admin of this repo would help us fix it.

**Comment by cgmaier** [comment]
hi all, our current understanding of this issue is that it's an issue with Xcode 16, and we're confident that you can safely ignore the warnings at submission. @JCsplash can you clarify with this is a blocker for you? 

some relevant links: 
https://forums.developer.apple.com/forums/thread/763731
https://github.com/mapbox/mapbox-maps-ios/issues/2233#issuecomment-2383192102
https://github.com/mapbox/mapbox-maps-ios/issues/2229#issuecomment-2320836023

**Comment by JCsplash** [comment]
Yes, this is definitely a blocker. I am unable to upload my build to the App Store. This is not just a warning. I need a fix for this in order to upload builds with Xcode 16. Would be awesome if you'd be able to help! @cgmaier

**Comment by JCsplash** [comment]
Actually @cgmaier the blocker is the invalid executable error on Xcode 16. When attempting to submit to the app store, im getting 
`Validation Failed: Invalid Executable. The executable 'Flare.app/Frameworks/GiphyUISDK.framework/GiphyUISDK'`

Is there any way you could help look into a fix for us?

**Comment by ALexanderLonsky** [comment]
@JCsplash 
I just uploaded multiple builds to the `App Store` using `Xcode 16.2`, with `SDK 2.2.14` and with `SDK 2.2.13`. All of them are currently in the `Ready to Submit` status.

The only warning I received for the builds is: `The archive did not include a dSYM for the GiphyUISDK.framework`. However, this is just a warning. I also double-checked my email from `App Store Connect`: `Version 1.0 (2) for Test Animations has completed processing`, and it confirms that processing is complete without listing any errors.

As @cgmaier pointed out, starting with `Xcode 16`, `Apple` has introduced stricter checks for native libraries included in frameworks during the `App Store` upload process. This results in warnings about missing debug symbols (`dSYM` files), even when these symbols are **intentionally** excluded to **reduce** the final `XCFramework's` binary size.

This behavior is expected for our framework. 

Could you please provide more details? We'd like to help, but I just can't reproduce your issue on my end.

I even tried validating the app, and the only message I received was this warning.

P.S. I’d like to note that I also tried adding the SDK using both `SPM` and `CocoaPods` and faced no issues apart from the warning.

**Comment by ALexanderLonsky** [comment]
Starting from version [2.2.15](https://github.com/Giphy/giphy-ios-sdk/releases/tag/2.2.15), we provide separate dSYM files to help you avoid related warnings. 
See the documentation [here](https://github.com/Giphy/giphy-ios-sdk/blob/main/dsym.md)

**Comment by JCsplash** [comment]
Hi @ALexanderLonsky - this is still an issue even after upgrading to 2.2.15. I'm on Xcode 16.2 and when i submit to the app store i get:

```The archive did not include a dSYM for the GiphyUISDK.framework with the UUIDs. Ensure that the archive's dSYM folder includes a DWARF file for GiphyUISDK.framework with the expected UUIDs.```

Can you please check the fix on your end again?

**Comment by JCsplash** [comment]
@ALexanderLonsky Please consider reopening this issue. Thanks!

**Comment by ALexanderLonsky** [comment]
@JCsplash, have you followed the [documentation](https://github.com/Giphy/giphy-ios-sdk/blob/main/dsym.md)?
